lib/util/git.js

```diff
@@ -25,6 +26,23 @@ const GOOD_ENV_VARS = new Set([
   'GIT_SSL_NO_VERIFY'
 ])
 
+const GIT_TRANSIENT_ERRORS = [
+  'remote error: Internal Server Error',
+  'fatal: Couldn\'t find remote ref ',
+  'The remote end hung up unexpectedly',
+  'Connection timed out',
+  'Operation timed out',
+  'Failed to connect to .* Timed out',
+  'Connection reset by peer',
+  'Couldn\'t resolve host'
```

Comment:
We should not retry on resolve failure, because that tends to be a sign of an offline state, and unlikely to be fixed on a retry. See [the errors make-fetch-happen retries on](https://github.com/zkat/make-fetch-happen/blob/latest/index.js#L18-L23).

Reply:
will remove, thank you.

Reply:
why this code still in repository???

https://github.com/zkat/pacote/blob/425c58d1f03b2c853dc0a5de1797b2804db8c481/lib/util/git.js#L33

Reply:
because this retry logic, it make npm install process will never end, when user can not access github by `http/https/ssh` protocol(like in Intranet CI environments).

if i dit not put `-ddd` arguments in command line, and i will never see what happend with npm.


```
$ node ./bin/npm-cli.js install -ddd github:hello/world
npm info it worked if it ends with ok
npm verb cli [ '/Users/xqin/.nvm/versions/node/v8.9.4/bin/node',
npm verb cli   '/Users/xqin/Desktop/Work/npm/bin/npm-cli.js',
npm verb cli   'install',
npm verb cli   '-ddd',
npm verb cli   'github:hello/world' ]
npm info using npm@6.8.0
npm info using node@v8.9.4
npm verb npm-session 146e905747a05a8f
npm sill install loadCurrentTree
npm sill install readLocalPackageData
npm sill pacote Retrying git command: ls-remote -h -t ssh://git@github.com/hello/world.git attempt # 2
npm sill pacote Retrying git command: ls-remote -h -t ssh://git@github.com/hello/world.git attempt # 3
npm sill pacote Retrying git command: ls-remote -h -t ssh://git@github.com/hello/world.git attempt # 4
npm sill pacote Retrying git command: ls-remote -h -t ssh://git@github.com/hello/world.git attempt # 5
npm sill pacote Retrying git command: ls-remote -h -t ssh://git@github.com/hello/world.git attempt # 6
npm sill pacote Retrying git command: ls-remote -h -t ssh://git@github.com/hello/world.git attempt # 7
npm sill pacote Retrying git command: ls-remote -h -t ssh://git@github.com/hello/world.git attempt # 8
npm sill pacote Retrying git command: ls-remote -h -t ssh://git@github.com/hello/world.git attempt # 9
npm sill pacote Retrying git command: ls-remote -h -t ssh://git@github.com/hello/world.git attempt # 10
⸨░░░░░░░░░░░░░░░░░░⸩ ⠦ rollbackFailedOptional: sill pacote Retrying git command: ls-remote -h -t ssh://git@github.com/hello/world.git attempt # 10
```

Reply:
@zkat  @dryganets